src/components/AppProvider/AppProvider.tsx

```diff
@@ -41,6 +42,7 @@ export interface AppProviderProps extends AppBridgeOptions {
   features?: Features;
   /** Inner content of the application */
   children?: React.ReactNode;
+  UNSTABLE_monorail?: MonorailObject; // eslint-disable-line babel/camelcase
```

Comment:
I'm not a fan of tieing this to monorail within the polaris-react codebase by this naming. Talking about internal shopify tooling feels like an odd leakage. Instead what if instead of passing in an instance of monorail we pass in a function that then calls monorail - or any other telemetry aggregator we might want in the future.

By pushing the responsibility of what telemetry producer is used to apps we let apps decide on how they want to handle the results, and any future changes should only need to be made in apps.

What if we set this up so consumers used this something like so:

```ts
// Or however web gets a monorail instance
const monorail = useMonorail();

function telemetryProducer(action: string, payload: Record<string, any> = {}) {
  // this example is lazy as the api of this function happens to map to monorail's api
 // but we could do clever things like batch or manipulate results here
  monorail.produce(action, payload);
}

const app = <AppProvider UNSTABLE_telemetry={telemetryProducer}>{/*...*/}</AppProvider>
```

Reply:
Using monorail specifically was an intentional decision to not commit us to supporting a generic telemetry that would work across the board.

This is a holdover from back when we were taking a different approach, making it generic here is simple enough.